packages/protocol/contracts/governance/Validators.sol

```diff
@@ -1131,4 +1131,29 @@ contract Validators is
       }
     }
   }
+
+  /**
+  * @notice Returns the group that `account` was a member of during `epochNumber`.
+  * @param account The account whose group membership should be returned.
+  * @param epochNumber The epoch number we are querying this account's membership at.
+  * @param index The index into the validator's history struct for their history at `epochNumber`.
+  * @return The group that `account` was a member of during `epochNumber`.
+  */
+  function groupMembershipInEpoch(address account, uint256 epochNumber, uint256 index)
+    external
+    view
+    onlyRegisteredContracts(canForceDeaffiliation)
+    returns (address)
+  {
+    require(isValidator(account) && epochNumber <= getEpochNumber());
+    MembershipHistory storage history = validators[account].membershipHistory;
+    require(
+      history.entries[index].epochNumber == epochNumber ||
+        (history.entries[index].epochNumber < epochNumber &&
```

Comment:
Are there any issues when any of these values are uninitialized? I don't think so, but please make sure you've thought about it carefully

Reply:
I addressed one that I missed which is if the epoch number really was 0, which I think _is possible_. 
I believe all uninitialized cases are handled, assuming history is updated correctly.

Reply:
For readability, consider the following:
```
require(index >= history.tail && history.numEntries > 0);
bool isExactMatch = history.entries[index].epochNumber == epochNumber;
bool isLastEntry = index.sub(history.tail) == history.numEntries.sub(1);
bool isWithinRange = history.entries[index].epochNumber < epochNumber && (history.entries[index.add(1)].epochNumber > epochNumber || isLastEntry);
require(isExactMatch || isWithinRange);
```